sys/include/xtimer.h

```diff
@@ -0,0 +1,442 @@
+ */
+#ifndef XTIMER
+#define XTIMER (0)
+#define XTIMER_CHAN (0)
+
+#if TIMER_0_MAX_VALUE == 0xffffff
+#define XTIMER_MASK 0xff000000
+#elif TIMER_0_MAX_VALUE == 0xffff
+#define XTIMER_MASK 0xffff0000
+#endif
+
+#endif
+/**
+ * @}
+ */
+
+#ifndef XTIMER_MASK
+/**
+ * @brief xtimer timer mask
+ *
+ * This value specifies the mask relative to 0xffffffff that the used timer
+ * counts to, e.g., 0xffffffff & ~TIMER_MAXVALUE.
+ *
+ * For a 16bit timer, the mask would be 0xFFFF0000, for a 24bit timer, the mask
+ * would be 0xFF000000. Don't set this for 32bit timers.
+ *
+ * This is supposed to be defined per-device in e.g., periph_conf.h.
+ */
+#define XTIMER_MASK 0
+#endif
+
+#ifndef XTIMER_USLEEP_UNTIL_OVERHEAD
+/**
+ * @brief xtimer_usleep_until overhead value
+ *
+ * This value specifies the time a xtimer_usleep_until will be late
+ * if uncorrected.
+ *
+ * This is supposed to be defined per-device in e.g., periph_conf.h.
+ */
+#define XTIMER_USLEEP_UNTIL_OVERHEAD 10
+#endif
+
+#if XTIMER_MASK
+extern volatile uint32_t _high_cnt;
+#endif
+
+/**
+ * @brief IPC message type for xtimer msg callback
+ */
+#define MSG_XTIMER 12345
+
+/**
+ * @brief returns the (masked) low-level timer counter value.
+ */
+static inline uint32_t _xtimer_now(void)
+{
+#ifdef XTIMER_SHIFT
+    return timer_read(XTIMER) << XTIMER_SHIFT;
+#else
+    return timer_read(XTIMER);
+#endif
+}
+
+/**
+ * @brief drop bits of a value that don't fit into the low-level timer.
+ */
+static inline uint32_t _mask(uint32_t val)
+{
+    return val & ~XTIMER_MASK;
+}
+
+/**
+ * @{
+ * @brief xtimer internal stuff
+ * @internal
+ */
+int _xtimer_set_absolute(xtimer_t *timer, uint32_t target);
+void _xtimer_set64(xtimer_t *timer, uint32_t offset, uint32_t long_offset);
+void _xtimer_sleep(uint32_t offset, uint32_t long_offset);
+static inline void xtimer_spin_until(uint32_t value);
+/** @} */
+
+static inline uint32_t xtimer_now(void)
+{
+#if XTIMER_MASK
+    return _xtimer_now() | _high_cnt;
+#else
+    return _xtimer_now();
+#endif
+}
+
+static inline void xtimer_spin_until(uint32_t value) {
+    while (_xtimer_now() > value);
+    while (_xtimer_now() < value);
+}
+
+static inline void xtimer_spin(uint32_t offset) {
+    offset = _mask(offset + _xtimer_now());
+    xtimer_spin_until(offset);
+}
+
+static inline void xtimer_usleep(uint32_t offset)
+{
+    _xtimer_sleep(offset, 0);
+}
+
+static inline void xtimer_usleep64(uint64_t microseconds)
+{
+    _xtimer_sleep((uint32_t) microseconds, (uint32_t) (microseconds >> 32));
+}
+
+static inline void xtimer_sleep(uint32_t seconds)
+{
+    xtimer_usleep64((uint64_t)seconds*1000000);
+}
+
+static inline void xtimer_nanosleep(uint32_t nanoseconds)
+{
+    _xtimer_sleep(nanoseconds/1000, 0);
+}
+
+/** @} */
+
+#if XTIMER_OVERHEAD + XTIMER_USLEEP_UNTIL_OVERHEAD > XTIMER_BACKOFF
+#warning (XTIMER_OVERHEAD + XTIMER_USLEEP_UNTIL_OVERHEAD > XTIMER_BACKOFF !!)
+#warning This will lead to underruns. Check if tests/xtimer_usleep_until runs through.
```

Comment:
As stated above: this pragma will cause an error (-Werror), but I'm not totally against it. But we might want to introduce other #warnings for unset variables that are expected to be set in `periph.conf` like an undefined `XTIMER_BACKOFF`.